### PR TITLE
Ethereum encoding: SigningOutput returns payload as well.

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

In Eth SigningOutput, alongside the full encoded+signed transaction return the transaction payload separately.  The payload is put together from inputs (e.g. Erc20, Erc721), now it will be returned separately.  Related to #1295 .

## Testing instructions

Eth unit tests.

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

**Review by vikmeup — APPROVED** [review]

File: src/proto/Ethereum.proto [hunk 1]
```diff
@@ -89,4 +89,7 @@ message SigningOutput {
     bytes v = 2;
     bytes r = 3;
     bytes s = 4;
+
+    // The payload part, returned separately
+    bytes payload = 5;
```
Comment (vikmeup) [inline src/proto/Ethereum.proto:94]: This is transaction.data correct?

**Review by vikmeup — APPROVED** [review]

File: src/proto/Ethereum.proto (hunk 1 above)
Comment (catenocrypt) [reply src/proto/Ethereum.proto:94]: Exactly.  For generic calls, data is supplied in input.  But for ERC20, 721, 1155 it is built from parameters supplied, so it is useful to return the built payload.

Comment (hewigovens) [reply src/proto/Ethereum.proto:94]: Let's call it input(naming in Ethereum rpc, like r,s,v) or just data

Comment (catenocrypt) [reply src/proto/Ethereum.proto]: Renamed to data.

**Review by hewigovens — APPROVED** [review]

**Review by KaycoinsInc — APPROVED** [review]

**Review by KaycoinsInc — APPROVED** [review]

**Pull request merged**